include/vstd/CLoggerBase.h

```diff
@@ -173,9 +173,8 @@ class DLL_LINKAGE CTraceLogger
 /// the first statement in the function. Logging traces via this macro have almost no impact when the trace is disabled.
 ///
 #define RAII_TRACE(logger, onEntry, onLeave)			\
-	std::unique_ptr<vstd::CTraceLogger> ctl00;						\
 	if(logger->isTraceEnabled())						\
-		ctl00 = std::make_unique<vstd::CTraceLogger>(logger, onEntry, onLeave);
+		std::unique_ptr<vstd::CTraceLogger> ctl00 = std::make_unique<vstd::CTraceLogger>(logger, onEntry, onLeave)
```

Comment:
If you want single-statement it should be
```cpp
std::unique_ptr<vstd::CTraceLogger> ctl00 = logger->isTraceEnabled() ? 
    std::make_unique<vstd::CTraceLogger>(logger, onEntry, onLeave) : 
    std::unique_ptr<vstd::CTraceLogger>();
```

Goal of this macro is to declare a variable that will be destroyed on leaving macro scope.
But with your change `ctl00` will be destroyed immediately after macro - on exit from `if` block